lua/obsidian/command.lua

```diff
@@ -523,7 +523,15 @@ command.follow = function(client, _)
     return
   end
 
-  local note_name = current_line:sub(open + 2, close - 1)
+  local note_name = current_line:sub(open, close)
+  if note_name:match('^%[.-%]%(.*%)$') then
+    -- transform markdown link to wiki link
+    note_name = note_name:gsub("^%[(.-)%]%((.*)%)$", "%2|%1")
```

Comment:
Here I'm changing markdown link to look like wiki link. I think that will be better for compatibility. If someone has custom  `note_id_func` it will still work nicely with markdown links.